### ISSUE TITLE
Test Framework

### ISSUE THREAD

**Issue by jawaff** [WaffleTime/KinectController#39]
We need a way to list out the test cases that haven't been started, are being worked on and have been completed. It would be really cool if we could show on our github repo which of the completed test cases are passing in the current build of the project as well -- although not necessarily needed. The list of test cases needs to be integrated into our github repo though -- whether it's done in the issues, the wiki or some other tool entirely.

The tests themselves would be a different matter. Our language options for the tests themselves would include Jython and just plain old Java from what I can currently tell. So some nice to use test framework for one of those languages is also going to be needed.

**Comment by jawaff** [comment]
It appears that travis-ci might be a cool tool to use -- although I've not looked into it all the way. It seems to only be for actually executing the different tests. So it doesn't do everything we need it to, but it would be nice for showing that our tests are passing on Github and automating our test execution.

It supports:
-Integration with github. 
-Test execution after each push to the repository.

tldr; travis-ci is cool and would automate the execution of our tests, but it does not have anything to do with writing the tests themselves.

**Comment by jawaff** [comment]
BEFORE looking into something like travis-ci or CircleCI, we should nail down the framework that we'll be using for actually creating tests. We also really need a way of managing the test cases that need to be tested -- the backup plan for this would be to handle them in github's issues under a test label.

Any test framework that we come up with can have any dependencies necessary, although I require a tutorial on how to setup the framework. Once the environment is setup, we'd need a single command/script that will execute the suite of tests within the framework. Then ideally, there would be some sort of .html file that would show which tests passed and which ones failed.

Test frameworks that I know would work are:
- robot-framework paired with Jython
  First example found: https://code.google.com/archive/p/robotframework/wikis/Installation.wiki
- TestNG paired with Java
- JUnit paired with Java

Note: If TestNG or JUnit are used, then I'd prefer if they were integrated with Gradle at some point -- Gradle is a build management tool that downloads our dependencies, injects them into the project, runs tests and eventually builds the project. Ideally I'd like to run the suite of tests using "gradle check" in a terminal -- that is the standard Gradle task that usually executes tests and all that cool stuff.

**Comment by jawaff** [comment]
I forgot that I already had an issue created for the addition of a test framework. Refer to issue #23
